packages/plugins/response-cache/src/index.ts

```diff
@@ -215,5 +230,40 @@ export default function useMeshResponseCache(
     cache: getCacheForResponseCache(options.cache),
     ttlPerType,
     ttlPerSchemaCoordinate,
+    onTtl({ ttl, context }) {
+      const ttlForThisContext = ttlByContext.get(context);
+      if (ttlForThisContext != null && ttlForThisContext < ttl) {
+        return ttlForThisContext;
+      }
+      return ttl;
+    },
   });
+  function checkTtl(context: GatewayContext, ttl: number) {
+    const ttlForThisContext = ttlByContext.get(context);
+    if (ttlForThisContext == null || ttl < ttlForThisContext) {
+      ttlByContext.set(context, ttl);
+    }
+  }
+  plugin.onFetch = function ({ executionRequest, context }) {
+    // If it is a subgraph request
+    if (executionRequest && context) {
+      return function onFetchDone({ response }) {
+        const cacheControlHeader = response.headers.get('cache-control');
+        if (cacheControlHeader != null) {
+          const parsedCacheControl = CacheControlParser.parse(cacheControlHeader);
+          if (parsedCacheControl['max-age']) {
```

Comment:
If the TTL is 0, this will not take it in account right ? Perhaps better to check against `null` instead ?

Reply:
I actually wanted to override their avoidance for caching but maybe it is not a good idea.